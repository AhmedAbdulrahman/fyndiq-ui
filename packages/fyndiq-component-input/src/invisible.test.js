import React from 'react'
import { shallow, mount } from 'enzyme'

import Invisible from './invisible'

describe('fyndiq-component-input Invisible', () => {
  it('should have the proper structure', () => {
    expect(shallow(<Invisible />)).toMatchSnapshot()
  })

  it('should have dirty class if the input is changed', () => {
    const component = shallow(<Invisible />)
    component
      .find('input')
      .simulate('change', { target: { value: 'newValue' } })

    expect(component.hasClass('invisibleWrapperDirty')).toBe(true)
  })

  it('should reset its value when pressing ESC key', () => {
    const component = mount(<Invisible value="test" />)

    // Change input field
    component
      .find('input')
      .simulate('change', { target: { value: 'newValue' } })

    // ESC key
    component.find('input').simulate('keydown', { keyCode: 27 })

    expect(component.find('input').prop('value')).toBe('test')
  })

  it('should update its value when the prop value changes', () => {
    const component = mount(<Invisible value="test" />)
    component.setProps({ value: 'newValue' })
    expect(component.find('input').prop('value')).toBe('newValue')
  })

  it('should not have a empty class when not empty', () => {
    const component = shallow(<Invisible value="test" />)

    expect(component.find('.invisibleInputEmpty').exists()).toBe(false)
  })

  it('should call onChange', () => {
    const spy = jest.fn()
    const component = mount(<Invisible value="test" onChange={spy} />)
    component
      .find('input')
      .simulate('change', { target: { value: 'newValue' } })

    component.simulate('submit', { preventDefault: jest.fn() })
    expect(spy).toHaveBeenCalledWith('newValue')
  })
})
