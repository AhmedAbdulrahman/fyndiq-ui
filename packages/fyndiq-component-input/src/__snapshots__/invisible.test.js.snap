// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fyndiq-component-input Invisible should have the proper structure 1`] = `
<form
  className="
          invisibleWrapper
          false
          
        "
  onSubmit={[Function]}
>
  <input
    className="
            invisibleInput
            invisibleInputEmpty
          "
    onChange={[Function]}
    onKeyDown={[Function]}
  />
  <button
    className="invisibleSubmit"
    type="submit"
  >
    <Pencil
      className="invisiblePencil"
    />
  </button>
</form>
`;
